push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java

```diff
@@ -209,4 +209,43 @@ public void messageSendFailed(ApnsNotification message, Throwable e) {
         // null if, why ever, there was no cert/passphrase
         return null;
     }
+
+    /**
+     * Configure the Gateway to the Apns servers.
+     * Default gateway and port can be override with respectively :
+     *  - aerogear.apns.push.host
+     *  - aerogear.apns.push.port
+     *
+     * Feedback gateway and port can be override with  respectively :
+     *  - aerogear.apns.feedback.host
+     *  - aerogear.apns.feedback.port
+     * @param iOSVariant
+     * @param builder
+     */
+    private void configureDestinations(iOSVariant iOSVariant, ApnsServiceBuilder builder) {
+        // pick the destination:
+        if (iOSVariant.isProduction()) {
+            builder.withProductionDestination();
+        } else {
+            builder.withSandboxDestination();
+        }
+
+        //Is the gateway host&port provided by a system property ?
```

Comment:
the above `withProductionDestination ()` or `withSandboxDestination()` is always executed, ok;

Only if these System properties are present, we override the URL/PORT. Not sure, but can we move that `System.getProperty` into a private method that is called like `checkForOverrideOfAPNsURLandPort()`? 

And its JavaDoc should be pretty clear, what's going on :-)